src/devices/sonoff.ts

```diff
@@ -649,6 +826,46 @@ const definitions: Definition[] = [
         description: 'Zigbee smart plug',
         extend: [onOff()],
     },
+    {
+        zigbeeModel: ['SWV'],
+        model: 'SWV',
+        vendor: 'SONOFF',
+        description: 'Zigbee smart water valve',
+        fromZigbee: [
+            fz.battery,
+            fz.flow,
+        ],
+        toZigbee: [],
+        ota: ota.zigbeeOTA,
+        exposes: [
+            e.battery(),
+            e.numeric('flow', ea.STATE).withDescription('Current water flow').withUnit('m³/h'),
+        ],
+        extend: [
+            enumLookup({
+                name: 'current_device_status',
+                lookup: {'normal state': 0, 'water shortage': 1, 'water leakage': 2, 'water shortage & water leakage': 3},
+                cluster: 0xFC11,
+                attribute: {ID: 0x500C, type: 0x20},
+                description: 'The water valve is in normal state, water shortage or water leakage',
+                access: 'STATE_GET',
+            }),
+            onOff({
+                powerOnBehavior: false,
+                skipDuplicateTransaction: true,
+                configureReporting: true,
+            }),
+            sonoffExtend.cyclicTimedIrrigation(),
+            sonoffExtend.cyclicQuantitativeIrrigation(),
+        ],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            const endpoint = device.getEndpoint(1);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['msFlowMeasurement']);
```

Comment:
Please create a new modern extend for this here https://github.com/Koenkk/zigbee-herdsman-converters/blob/bc0506cd8e40d7c4ec5b108cc2c652c7b184dead/src/lib/modernExtend.ts#L743